controllers/propagator/propagation.go

```diff
@@ -107,10 +107,47 @@ func getRetryOptions(logger logr.Logger, retryMsg string) []retry.Option {
 	return common.GetRetryOptions(logger, retryMsg, uint(attempts))
 }
 
+func (r *PolicyReconciler) deletePolicy(plc *policiesv1.Policy) error {
+	// #nosec G601 -- no memory addresses are stored in collections
+	err := r.Delete(context.TODO(), plc)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		log.Error(
+			err,
+			"Failed to delete the replicated policy",
+			"name", plc.GetName(),
+			"namespace", plc.GetNamespace(),
+		)
+
+		return err
+	}
+
+	return nil
+}
+
+type policyDeleter interface {
+	deletePolicy(instance *policiesv1.Policy) error
+}
+
+type deletionResult struct {
+	Identifier string
```

Comment:
It seems that `Identifier` is not used, so you could simplify the result channel to just accepting `error` objects.